lib/rails_admin/adapters/active_record.rb

```diff
@@ -178,11 +178,78 @@ def range_filter(min, max)
 
         def build_statement_for_type
           case @type
-          when :boolean                   then build_statement_for_boolean
-          when :integer, :decimal, :float then build_statement_for_integer_decimal_or_float
-          when :string, :text             then build_statement_for_string_or_text
-          when :enum                      then build_statement_for_enum
-          when :belongs_to_association    then build_statement_for_belongs_to_association
+            when :jsonb, :json then
+              build_statement_for_jsonb
+            when :boolean then
+              build_statement_for_boolean
+            when :integer, :decimal, :float then
+              build_statement_for_integer_decimal_or_float
+            when :string, :text then
+              build_statement_for_string_or_text
+            when :enum then
+              build_statement_for_enum
+            when :belongs_to_association then
+              build_statement_for_belongs_to_association
+          end
+        end
+
+        def build_statement_for_jsonb
+          return unless ar_adapter == 'postgresql'
```

Comment:
A cleaner way would be to extend the current AR adapter into a specialised postgres 9.4+ adapter and write logic to have it selected automatically.
Then with that we unlock all the json/jsonb options for rails_admin UI.